### PR TITLE
E2E Tests: Fix failing image e2e test by waiting for required element

### PR DESCRIPTION
**Pull request by glendaviesnz**
## Description
The image block e2e tests are consistently failing due to element not being visible when attempting to click. This fix adds a `page.waitForSelector` to ensure element is visible before attempting the click.

There is also an update to the test for `Gutenberg_REST_Templates_Controller_Test` that updates the `test_create_item` test to expect the author id in the response to match the id used to create the resource (`self::$admin_id`).

## How has this been tested?
Run `npm run test-e2e specs/editor/blocks/image.test.js` locally and make sure tests pass.

### CODE REVIEW

**Review by ramonjd — APPROVED** [review]
Thank you for fixing this @glendaviesnz 


Screen Shot 2021-11-30 at 11 02 09 am

Tests pass locally and on CI!

**Comment by ramonjd** [comment]
I also tested 23db42264d1029361fa93cc7ec3fb3520492872f, which reinstates the test from https://github.com/WordPress/gutenberg/pull/36446

They're passing locally for me. 🤞 for the CI results.

**Review by aaronrobertshaw — APPROVED** [review]
LGTM 👍 

The latest `waitForImage` approach tests well for me and passes reliably locally. For what it's worth, the previous `waitForSelector` approach failed consistently for me locally.

**Review by noisysocks — APPROVED** [review]
Thanks for this!

**Comment by Mamaduka** [comment]
Thanks for working on this, @glendaviesnz 🌟 

I'm going to restart e2e tests and merge this once all checks are green. Then, we can address other flaky tests separately.

**Comment by aaronrobertshaw** [comment]
> I'm going to restart e2e tests and merge this once all checks are green. Then, we can address other flaky tests separately.

@andrewserong and I have been testing this locally and finding the remaining image e2es are failing intermittently.  Every other run to be precise. We might need a little luck to get all green checks 🤞

**Comment by Mamaduka** [comment]
> We might need a little luck to get all green checks

Or we could use administrator privileges and merge it anyways. It only will make the current status of e2e tests better 😄

**Comment by Mamaduka** [comment]
This error is the current reason for test failure. I've seen it a dozen times in the last couple of days but cannot reproduce it locally.

## Error
```
thrown: Response {
      "size": 0,
      "timeout": 0,
      Symbol(Body internals): Object {
        "body": PassThrough {
          "_events": [Object],
          "_eventsCount": 2,
          "_maxListeners": undefined,
          "_readableState": [ReadableState],
          "_writableState": [WritableState],
          "allowHalfOpen": true,
          Symbol(kCapture): false,
          Symbol(kTransformState): [Object],
        },
        "disturbed": false,
        "error": null,
      },
      Symbol(Response internals): Object {
        "counter": 0,
        "headers": Headers {
          Symbol(map): [Object],
        },
        "status": 400,
        "statusText": "Bad Request",
        "url": "http://localhost:8889/wp-admin/admin-ajax.php?action=rest-nonce",
      },
    }
```

**Comment by aaronrobertshaw** [comment]
> Or we could use administrator privileges and merge it anyways. It only will make the current status of e2e tests better 😄

I wasn't looking to push that option given locally I'm getting more failures on this branch than before. Given the inconsistency, if merging this lets us narrow the investigation, I'm fine with that 🙂 

<details>
    <summary>Here's some of the errors we're seeing locally</summary>
    
Screen Shot 2021-11-30 at 3 07 34 pm

Screen Shot 2021-11-30 at 3 20 53 pm
    
image
</details>

**Comment by andrewserong** [comment]
I _think_ I have an idea for what could be causing the failing `Should reset dimensions on change URL` to fail intermittently (or one of the factors at play, at least). It seems that locally the test is running too quickly, so that typing in the replacement URL, the whole URL doesn't get typed out, or the existing URL doesn't get removed. Adding a short delay appears to get it passing for me locally.

I'll push a commit shortly.

Edit: pushed in https://github.com/WordPress/gutenberg/pull/36982/commits/2eb00f2606d5a7fe177fbe9ab67116d6b8ee4a4f

The hunch is that there's a couple of things at play. I suspect that doing things like the following might cause an issue for React controlled components (because we're updating the value of input field from outside of React):

```
await page.evaluate(
	() =>
		( document.querySelector(
			'.block-editor-url-input__input'
		).value = '' )
);
```

So, it could be more appropriate for us to try to clear fields as a user would? Then, the other thing at play is how quickly the UI will let us type. Adding in a delay appears to help reliability when running the tests locally, but as Aaron mentions, we're still experiencing intermittent failures 🤔

**Comment by aaronrobertshaw** [comment]
> I think I have an idea for what could be causing the failing Should reset dimensions on change URL to fail intermittently (or one of the factors at play, at least)

Initially, this looked like it fixed the issue with `Should reset dimensions on change URL`. On some runs though, a similar error occurs where it appears the URL input doesn't get cleared properly. 

I also still got the `allows changing aspect ratio using the crop tools` failure again.

File: packages/e2e-tests/specs/editor/blocks/image.test.js [hunk 1]
```diff
@@ -337,15 +341,15 @@ describe( 'Image', () => {
...
-			() =>
-				( document.querySelector(
-					'.block-editor-url-input__input'
-				).value = '' )
-		);
 
+		// Clear the input field. Delay added to account for typing delays.
+		const inputField = await page.$( '.block-editor-url-input__input' );
+		await inputField.click( { clickCount: 3, delay: 100 } );
+		await page.keyboard.press( 'Backspace', { delay: 100 } );
```
Comment (aaronrobertshaw) [inline packages/e2e-tests/specs/editor/blocks/image.test.js:348]: Keeping the total delay overall the same but just using all of it to ensure the selection of the input field text has the e2es passing more reliably for me.

```suggestion
		await inputField.click( { clickCount: 3, delay: 200 } );
```

**Comment by aaronrobertshaw** [comment]
@Mamaduka I haven't been able to get the checks green despite passing e2es locally. We might need to merge this with admin privileges after all so at least the PHP unit tests are fixed.

I've run out of time today but I've dropped a comment pointing to this PR in the `core-editor` Slack channel in case someone can pick this up.

**Comment by Mamaduka** [comment]
Sounds good, @aaronrobertshaw 👍

**Comment by Mamaduka** [comment]
I ran the E2E tests locally on fresh `wp-env` installation 15 times using a loop, and it passed every time.

```
for i in {1..15}; do npm run test-e2e -- packages/e2e-tests/specs/editor/blocks/image.test.js; done
```

**Comment by talldan** [comment]
> This error is the current reason for test failure. I've seen it a dozen times in the last couple of days but cannot reproduce it locally.

This is curious. Seems to be the whole test suite failing to run. Is it some kind of issue with authentication since `http://localhost:8889/wp-admin/admin-ajax.php?action=rest-nonce` is the url returning the `400` status code?

**Comment by Mamaduka** [comment]
> Is it some kind of issue with authentication since

That's also my best guess, maybe issue with auth cookies.

File: packages/e2e-tests/specs/editor/various/mentions.test.js [hunk 2]
```diff
@@ -10,7 +10,9 @@ import {
 	pressKeyTimes,
 } from '@wordpress/e2e-test-utils';
 
-describe( 'autocomplete mentions', () => {
+// Something in WP 5.9 RC has broken the creation of new users so disabling until
+// that is fixed
+describe.skip( 'autocomplete mentions', () => {
```
Comment (talldan) [inline packages/e2e-tests/specs/editor/various/mentions.test.js]: This is passing in https://github.com/WordPress/gutenberg/pull/36987. Might be worth enabling it again 😄

**Comment by talldan** [comment]
> That's also my best guess, maybe issue with auth cookies.

The unsual thing is that it's only that image test and I see nothing special about it. Other than it possibly being the first test run in the series.

**Comment by talldan** [comment]
I merged the PHP test fix separately in https://github.com/WordPress/gutenberg/pull/36987.

Also rebased this and re-enabled the autocomplete tests as they now seem to be passing in trunk.

**Pull request merged**